zhmcclient/_adapter.py

```diff
@@ -0,0 +1,178 @@
+Objects of this class are not provided when the CPC is not in DPM mode.
+
+An adapter is always contained in a CPC.
+"""
+
+from __future__ import absolute_import
+
+from ._manager import BaseManager
+from ._resource import BaseResource
+
+__all__ = ['AdapterManager', 'Adapter']
+
+
+class AdapterManager(BaseManager):
+    """
+    Manager object for Adapters. This manager object is scoped to the
+    adapters of a particular CPC.
+
+    Derived from :class:`~zhmcclient.BaseManager`; see there for common methods
+    and attributes.
+    """
+
+    def __init__(self, cpc):
+        """
+        Parameters:
+
+          cpc (:class:`~zhmcclient.Cpc`):
+            CPC defining the scope for this manager object.
+        """
+        super(AdapterManager, self).__init__(cpc)
+
+    @property
+    def cpc(self):
+        """
+        :class:`~zhmcclient.Cpc`: Parent object (CPC) defining the scope for
+        this manager object.
+        """
+        return self._parent
+
+    def list(self, full_properties=False):
+        """
+        List the adapters in scope of this manager object.
+
+        Parameters:
+
+          full_properties (bool):
+            Controls whether the full set of resource properties should be
+            retrieved, vs. only the short set as returned by the list
+            operation.
+
+        Returns:
+
+          : A list of :class:`~zhmcclient.Adapter` objects.
+
+        Raises:
+
+          :exc:`~zhmcclient.HTTPError`
+          :exc:`~zhmcclient.ParseError`
+          :exc:`~zhmcclient.AuthError`
+          :exc:`~zhmcclient.ConnectionError`
+        """
+        cpc_uri = self.cpc.get_property('object-uri')
+        adapters_res = self.session.get(cpc_uri + '/adapters')
+        adapter_list = []
+        if adapters_res:
+            adapter_items = adapters_res['adapters']
+            for adapter_props in adapter_items:
+                adapter = Adapter(self, adapter_props['object-uri'],
+                                  adapter_props)
+                if full_properties:
+                    adapter.pull_full_properties()
+                adapter_list.append(adapter)
+        return adapter_list
+
+    def create_hipersocket(self, properties):
+        """
+        Create and configures a HiperSockets adapter
+        with the specified resource properties.
+
+        Parameters:
+
+          properties (dict): Properties for the new adapter.
+
+        Returns:
+
+          string: The resource URI of the new adapter.
+
+        Raises:
+
+          :exc:`~zhmcclient.HTTPError`
+          :exc:`~zhmcclient.ParseError`
+          :exc:`~zhmcclient.AuthError`
+          :exc:`~zhmcclient.ConnectionError`
+        """
+        cpc_uri = self.cpc.get_property('object-uri')
+        result = self.session.post(cpc_uri + '/adapters', body=properties)
+        return result['object-uri']
+
+
+class Adapter(BaseResource):
+    """
+    Representation of an Adapter.
```

Comment:
Clarify that this is a physical adapter on the CPC (or a part of that?), or a HiperSockets switch.